inspirehep/modules/workflows/views/callback.py

```diff
@@ -104,7 +103,32 @@ def _find_and_continue_workflow(workflow_id, recid, result=None):
             'message': 'workflow with id %s not found.' % workflow_id,
         }
 
-    return {'status': 'succeeded'}
+    return {
+        'status': 'succeeded',
+        'message': 'workflow with id %s continued.' % workflow_id,
+    }
+
+
+def _put_workflow_in_error_state(workflow_id, error_message, result):
+    workflow_object = workflow_object_class.get(workflow_id)
+    if not workflow_object:
+        current_app.logger.error(
+            'No workflow object with the id %s could be found.',
+            workflow_id,
+        )
+        return {
+            'status': 'failed',
+            'message': 'workflow with id %s not found.' % workflow_id,
+        }
+
+    workflow_object.status = ObjectStatus.ERROR
+    workflow_object.extra_data['callback_result'] = result
+    workflow_object.extra_data['_error_message'] = error_message
+    workflow_object.save()
```

Comment:
In all other occurrences of `workflow_object.save()` it is always followed by `db.session.commit()` (except on this file).

I would say it is needed since `save()` just adds to the session.